src/NUnitTestAdapter/TestConverter.cs

```diff
@@ -221,6 +221,30 @@ private VSTestResult GetBasicResult(XmlNode resultNode)
 
             vsResult.ComputerName = Environment.MachineName;
 
+            foreach (var output in outputNodes)
+            {
+                var stream = output.GetAttribute("stream");
+                if (string.IsNullOrEmpty(stream))
+                {
+                    continue;
+                }
+
+                var text = output.InnerText;
+                string category;
+                switch (stream.ToLower())
+                {
+                    case "error":
+                        category = TestResultMessage.StandardErrorCategory;
+                        break;
+
+                    default:
+                        category = TestResultMessage.StandardOutCategory;
+                        break;
+                }
+
+                vsResult.Messages.Add(new TestResultMessage(category, text));
+            }
+
             XmlNode outputNode = resultNode.SelectSingleNode("output");
             if (outputNode != null)
                 vsResult.Messages.Add(new TestResultMessage(TestResultMessage.StandardOutCategory, outputNode.InnerText));
```

Comment:
Can you add a comment clarifying why we are doing both things, still using `SelectSingleNode("output")` here and adding a final message?

Reply:
Done